### ISSUE TITLE
Raise diagnostic for duplicate `ParamSpec` components arguments

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ty#3745]
Ref https://github.com/astral-sh/ty/issues/1861

The following should raise a diagnostic given that `*args` and `**kwargs` are provided multiple times:

```py
from typing import Callable

def duplicate1[**P](f: Callable[P, None], *args: P.args, **kwargs: P.kwargs) -> None:
    f(*args, *args, **kwargs)          # should be rejected
    f(*args, **kwargs, **kwargs)       # should be rejected
```

Note that this is specific to when `ParamSpec`s are involved and should _not_ be raised for other non-ParamSpec cases like the following:

```py
def gradual(*args, **kwargs): ...

def works(*args, **kwargs) -> None:
    gradual(*args, *args, **kwargs)
    gradual(*args, **kwargs, **kwargs)
```

**Comment by AlexWaygood** [comment]
> Note that this is specific to when `ParamSpec`s are involved and should be raised for other non-ParamSpec cases like the following:

Should that say "should _not_ be raised"?

**Comment by mvanhorn** [comment]
Implemented in https://github.com/astral-sh/ty/pull/3796 - feat: Raise a diagnostic for duplicate ParamSpec component arguments at call sit.
